test/GitcoinGovernor.t.sol

```diff
@@ -301,137 +289,177 @@ contract GitcoinGovernorAlphaPostProposalTest is GitcoinGovernorProposalTestHelp
       governorAlpha.castVote(_newProposalId, true);
     }
     vm.roll(_endBlock + 1);
+
+    if (!isGovernorAlphaAdmin) {
+      vm.expectRevert("Timelock::queueTransaction: Call must come from admin.");
+      governorAlpha.queue(_newProposalId);
+      return;
+    }
+
     governorAlpha.queue(_newProposalId);
     vm.roll(block.number + 1);
     (,, uint256 _eta,,,,,,) = governorAlpha.proposals(_newProposalId);
     vm.warp(_eta + 1);
+
     governorAlpha.execute(_newProposalId);
 
     // Ensure the new proposal is now executed
-    uint8 _state = governorAlpha.state(_newProposalId);
-    assertEq(_state, EXECUTED);
+    assertEq(governorAlpha.state(_newProposalId), EXECUTED);
+  }
+
+  function testFuzz_OldGovernorSendsETHAfterProposalIsDefeated(uint256 _amount, address _receiver)
+    public
+  {
+    _assumeReceiver(_receiver);
+
+    // Counter-intuitively, the Governor must hold the ETH, not the Timelock.
+    // See the deployed GovernorAlpha, line 227:
+    //   https://etherscan.io/address/0xDbD27635A534A3d3169Ef0498beB56Fb9c937489#code
+    // The governor transfers ETH to the Timelock in the process of executing
+    // the proposal. The Timelock then just passes that ETH along.
```

Comment:
Huh interesting! What if the timelock holds ETH? It's just stuck there? I recall Compound had a similar issue and had to upgrade the timelock, but I think this one is non-upgradeable.
- https://www.comp.xyz/t/enable-transfer-eth-from-timelock/2689
- https://compound.finance/governance/proposals/115

Reply:
Oh, yeah that is interesting!

Thankfully, if the timelock holds ETH it can still transfer it. During execution, it [uses](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c8b466b7b55aee3967af21cb3c659ec07e93021d/contracts/mocks/compound/CompTimelock.sol#L162) the `value` [provided by the caller](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c8b466b7b55aee3967af21cb3c659ec07e93021d/contracts/mocks/compound/CompTimelock.sol#L139). So if the caller provides a value, it will send that much ETH. This is what happens in our tests for the GovernorBravo upgrade, i.e. we [deal directly to the TIMELOCK](https://github.com/gitcoinco/Alpha-Governor-Upgrade/pull/16/files#diff-0f96982af50a31c5aafa2f8d38e1213fb796bf8bcad9cb4620670de9d477373dR695) and then show it can transfer the ETH.

The issue here is really just confined to GovernorAlpha. This is [line 227](https://etherscan.io/address/0xDbD27635A534A3d3169Ef0498beB56Fb9c937489#code), which calls `executeTransaction` on the Timelock (with some indentation added so it doesn't overflow):

```solidity
timelock.executeTransaction.value(proposal.values[i])(
  proposal.targets[i],
  proposal.values[i],
  proposal.signatures[i],
  proposal.calldatas[i],
  proposal.eta
);
```

It sends `proposal.values[i]` at the same time as it passes it as the second argument. So with GovernorAlpha you can't have the Timelock execute a transaction that sends ETH unless you also send the same amount of ETH to the Timelock.